src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressType.php

```diff
@@ -142,6 +142,17 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     $orderData['billingAddress'] = $orderData['shippingAddress'];
                 }
 
+                $event->setData($orderData);
+            })
+            ->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) use ($options): void {
+                $orderData = $event->getData();
+
+                if ($options['customer'] instanceof CustomerInterface && array_key_exists('customer', $orderData)) {
+                    // Logged-in user try to submit customer while reloading page after login
+                    unset($orderData['customer']);
+                    $event->getForm()->remove('customer');
+                }
+
                 $event->setData($orderData);
             })
```

Comment:
I think the issue here is that the customer field is no longer added by the other listener, so this could be written as:

```suggestion
            ->addEventListener(FormEvents::PRE_SUBMIT, static function (FormEvent $event): void {
                $orderData = $event->getData();
                $form = $event->getForm();

                if (array_key_exists('customer', $orderData) && !$form->has('customer')) {
                    // Logged-in user try to submit customer while reloading page after login
                    unset($orderData['customer']);
                    $event->setData($orderData);
                }
            })
```

Reply:
Make sense ! Thanks :slightly_smiling_face: